app/src/test/contrib/actions/CoverageActionsTests.ts

```diff
@@ -0,0 +1,80 @@
+import { expect } from "chai";
+
+import { Sandbox } from "../../Sandbox";
+import { coverageActionCreators } from "../../../main/contrib/actions/coverageActionCreators";
+import { CoverageService } from "../../../main/contrib/services/CoverageService";
+import {Coverage, CoverageTypes} from "../../../main/contrib/actionTypes/CoverageTypes";
+import {createMockStore} from "../../mocks/mockStore";
+import {mockCoverageSet, mockDemographicDataset, mockScenario, mockTouchstone} from "../../mocks/mockModels";
+import {ScenarioAndCoverageSets} from "../../../main/shared/models/Generated";
+import {statePropsMapHelper} from "../../../main/contrib/helpers/statePropsMapHelper";
+
+describe("Coverage actions tests", () => {
+    const sandbox = new Sandbox();
+
+    const testCoverageSet = mockCoverageSet();
+    const testScenario = mockScenario();
+    const testScenarioAndCoverageSet: ScenarioAndCoverageSets = {
+        coverage_sets: [testCoverageSet],
+        scenario: testScenario
+    };
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it("data sets fetched", (done) => {
+        const store = createMockStore({});
+        sandbox.setStubFunc(statePropsMapHelper, "getResponsibilityIds", ()=>{
+            return Promise.resolve({groupId: "g1", touchstoneId: "t1", scenarioId: "s1"});
```

Comment:
Why does this need to be a promise? The original function is synchronous.

Reply:
this was copy paste from somewhere, will need to be fixed, yes

Reply:
This still needs to be fixed, I think?